### PR TITLE
Performance controller

### PR DESCRIPTION
**Pull request by DelphineHelie**
Création du performance controller
White-listing des pages auquelles le visiteur doit avoir accès
Crud methodes 
Index view en cours

### CODE REVIEW

File: app/controllers/performances_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class PerformancesController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:index, :show, :book]
```
Comment (Jucando) [inline app/controllers/performances_controller.rb]: should delete book and put it in Reservatinos controller (method new/create)

File: app/controllers/performances_controller.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  def index
+    @performances = Performance.all
+  end
+
+  def show
```
Comment (DelphineHelie) [inline app/controllers/performances_controller.rb]: add redirect_to new_performance_reservation_path

File: app/controllers/performances_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    @performance = Performance.find(params[:id])
+  end
+
+  def new
+    @performance = Performance.new(performance_params)
```
Comment (DelphineHelie) [inline app/controllers/performances_controller.rb]: sup (performance_params)

File: app/controllers/performances_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  def book
+
+  end
+
+  private
+
+  def performance_params
+    # *Strong params*: You need to *whitelist* what can be updated by the user
+    # Never trust user data!
+    params.require(:performance).permit(:title, :description, :price, :is_visible, :user)
```
Comment (DelphineHelie) [inline app/controllers/performances_controller.rb]: delete user
add category

File: app/controllers/pages_controller.rb [hunk 5]
```diff
@@ -1,6 +1,15 @@
 class PagesController < ApplicationController
-  skip_before_action :authenticate_user!, only: [:home]
+  skip_before_action :authenticate_user!, only: [:home, :index, :show, :book]
```
Comment (victorlacombe) [inline app/controllers/pages_controller.rb]: @DelphineHelie 
delete index, show, book methods and related skip_before_action

File: app/models/performance.rb [hunk 6]
```diff
@@ -1,3 +1,4 @@
 class Performance < ApplicationRecord
   belongs_to :user
+  has_many :media
```
Comment (victorlacombe) [inline app/models/performance.rb]: has_many :reservations

**Pull request merged**